segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/DeltaIteratorState.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.tables;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.common.util.ArrayView;
+import io.pravega.segmentstore.contracts.tables.IteratorState;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.ToString;
+
+import java.io.IOException;
+
+/**
+ * Represents the state of a resumable iterator.
```

Comment:
"resumable delta iterator"


Let's differentiate between `IteratorState` and `DeltaIteratorState`. The Javadoc should explain what is different about this one.